src/client/app/shared/models/game.model.ts

```diff
@@ -32,9 +32,11 @@ export class GameModel extends Model {
   }
 
   completeGame(time: number, text: string) {
-    const action = GameActions.completeGame(time, text);
-    this._store.dispatch(action);
-    this.performAsyncAction(action)
-      .subscribe(() => console.log('Done!'));
+    this._store.dispatch(GameActions.completeGame(time, text));
+    this.performAsyncAction(GameActions.completeGame(new Date(), text))
```

Comment:
I refactored it so that the same action is used for both store and async services. I am not passing Date.now() but the timer time, and in the restful gateway I compare to 0 or last timer time. The other async service was not using the time, so it is not affected by sending a different value.